torch/_dynamo/variables/builder.py

```diff
@@ -682,6 +682,11 @@ def build_key_value(i, k, v):
         elif istype(value, frozenset) and (
             ConstantVariable.is_literal(x) for x in value
         ):
+            # TODO: the `is_literal` check above is futile, because we are
+            # missing a `all` call around this generator object; this means we
+            # always wrap a fronzent with 1 layer of `ConstantVariable`, which
+            # interacts in a subtle way with `ConstantVariable.create`.
+            #
```

Comment:
I tried to remove this branch in favor of `ConstantVariable.create`, but failed because
1. The branch was added for `test_frozenset_torch_func_contains`, which contains a frozenset of function objects.
2. `ConstantVariable.is_literal` fails for function objects
3. Even if we fix (2), we don't want `ConstantVariable.create` to end up using `ConstantVariable` for the item function, because it might get loaded and called later, as in `test_frozenset_torch_func_contains`.

It's unclear whether the lack of `all(...)` call here is intended. I _feel_ like it's a bug that happens to work for `test_frozenset_torch_func_contains`.

But let's say we fix (3) as well by using `LazyVariableTracker` or `VariableTracker.build` for the nested items. Then why do we even need `ConstantVariable` at all, why not just replace all uses of it with `LazyVariableTracker` or `VariableTracker.build`...?

Reply:
This looks like a bug to me, we should add the `all()` and fix and bugs that introduces.

Reply:
Addressed in #141504.